### PR TITLE
chore: lint overhaul

### PR DESCRIPTION
**Pull request by Benjozork**
## Summary of Changes

- Moves lint to a simpler (and stricter) setup with `eslint:recommended`, `plugin:prettier/recommended` and `plugin:@typescript-eslint/recommended`
- Fixes lint failures introduced by the above

### CODE REVIEW

**Review by frankkopp — APPROVED** [review]
Checking line by line is of course impossible ;)
But by spot checking changes it seems this worked well. 
Also no issues in the Installer itself.

File: src/renderer/components/SettingsSection/Developer.tsx [hunk 1]
```diff
@@ -1,53 +1,50 @@
...
+  <div className="flex flex-row items-center justify-between py-3.5">
+    <p className="m-0">{name}</p>
+    {children}
+  </div>
 );
 
-const index = (): JSX.Element => {
-    const [configDownloadUrl, setConfigDownloadUrl] = useSetting<string>('mainSettings.configDownloadUrl');
-    const [configDownloadUrlValid, setConfigDownloadUrlValid] = useState<boolean>(false);
+export const DeveloperSettings = (): JSX.Element => {
```
Comment (FoxtrotSierra6829) [inline src/renderer/components/SettingsSection/Developer.tsx:11]: Could you please keep unrelated changes out of a linting PR?

**Comment by FoxtrotSierra6829** [comment]
Still unsure how to feel about the change from 4-space to 2-space indentation. Are we changing this on all FBW repositories? I saw you did in [flybywiresim/simbridge](https://github.com/flybywiresim/simbridge). Maybe something to discuss with other devs?

**Comment by frankkopp** [comment]
> Still unsure how to feel about the change from 4-space to 2-space indentation. Are we changing this on all FBW repositories? I saw you did in [flybywiresim/simbridge](https://github.com/flybywiresim/simbridge). Maybe something to discuss with other devs?

Yes - all repos should have the same linting.

Also 2 spaces are imho good - the C++ code already has this for example. 
I would vote for 2 vs. 4.

**Comment by FoxtrotSierra6829** [comment]
> > Still unsure how to feel about the change from 4-space to 2-space indentation. Are we changing this on all FBW repositories? I saw you did in [flybywiresim/simbridge](https://github.com/flybywiresim/simbridge). Maybe something to discuss with other devs?
> 
> Yes - all repos should have the same linting.
> 
> Also 2 spaces are imho good - the C++ code already has this for example. I would vote for 2 vs. 4.

I find more 4 spaces easier to read over multiple nested blocks. The main advantage to me would be that the deeper blocks have less of an issue to exceed the maximum line length.

File: src/renderer/components/SettingsSection/Developer.tsx (hunk 1 above)
Comment (Benjozork) [reply src/renderer/components/SettingsSection/Developer.tsx:11]: That is a linting change. It fixes the function not being considered a react component, which triggers a lint rule regarding hooks.

**Comment by Benjozork** [comment]
Maybe that will discourage excessive nesting then 😄 

https://www.youtube.com/watch?v=CFRhGnuXG-4

**Comment by tracernz** [comment]
4 is definitely a lot more readable, with the side benefit of discouraging excessive nesting. If only we could all use tabs thus could set our own preference in our editor of choice.

**Comment by derfelix54** [comment]
> Hey @Benjozork , after those changes I cannot build the master branch locally with `npm run package` anymore. Is there something I have to do, not mentioned in the readme?

Update: source-map is missing in the package.json. After npm install it built.
